frontend/javascripts/oxalis/view/plane_view.js

```diff
@@ -137,6 +143,69 @@ class PlaneView {
     }
   }
 
+  performIsosurfaceHitTest(): ?THREE.Vector3 {
+    const storeState = Store.getState();
+    const SceneController = getSceneController();
+    const { isosurfacesRootGroup } = SceneController;
+    const tdViewport = getInputCatcherRect(storeState, "TDView");
+    const { mousePosition, hoveredIsosurfaceId } = storeState.temporaryConfiguration;
+
+    if (mousePosition == null) {
+      return null;
+    }
+
+    // Outside of the 3D viewport, we don't do isosurface hit tests
+    if (storeState.viewModeData.plane.activeViewport !== OrthoViews.TDView) {
+      if (hoveredIsosurfaceId !== 0) {
+        // Reset hoveredIsosurfaceId if we are outside of the 3D viewport,
+        // since that id takes precedence over the shader-calculated cell id
+        // under the mouse cursor
+        Store.dispatch(updateTemporarySettingAction("hoveredIsosurfaceId", 0));
+      }
+      return null;
+    }
+
+    // Perform ray casting
+    const mouse = new THREE.Vector2(
+      (mousePosition[0] / tdViewport.width) * 2 - 1,
+      ((mousePosition[1] / tdViewport.height) * 2 - 1) * -1, // y is inverted
+    );
+
+    raycaster.setFromCamera(mouse, this.cameras[OrthoViews.TDView]);
+    const intersections = raycaster.intersectObjects(isosurfacesRootGroup.children, true);
```

Comment:
Why do we need to intersect recursively? Is this because the segments for one id are in a group or is there another reason?

Reply:
Yes, it's because the segments are in a group and the group is only traversed if `recursive` is true. I'll add a comment!